### PR TITLE
chore: fix dockerfile FROM statement

### PR DESCRIPTION
**Pull request by insumity**
**Problem**
Trying to run the e2e tests locally would give:
```
OCI runtime exec failed: exec failed: unable to start container process: exec /usr/local/bin/hermes: no such file or directory: unknown
exit status 1
make: *** [test-e2e-short] Error 1
```

**Solution**
Fixed by adding `--platform=linux/amd64` in the relevant `FROM` statement.
Also, added the `Dockerfile` in the test GitHub actions because a change there might lead to the tests failing.

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

**Comment by p-offtermatt** [comment]
Let's also add the same FROM in Dockerfile.gaia, right?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -192,6 +192,8 @@ jobs:
             **/go.sum
             **/Makefile
             Makefile
+            Dockerfile
+            Dockerfile.gaia
```
Comment (p-offtermatt) [inline .github/workflows/test.yml:196]: Maybe do Dockerfile
and Dockerfile.*
e.g. if we get a Dockerfile.test at some point, it's covered.
(or even just Dockerfile*, instead)

**Review by p-offtermatt — APPROVED** [review]
LGTM, a nit. I also added a modification in Dockerfile.gaia, please check it out and revert if you think it's not correct

**Review by bermuell — APPROVED** [review]
LGTM, pls address comments from @p-offtermatt

**Pull request merged**